querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQueryFactory.java

```diff
@@ -90,4 +91,61 @@ public final Connection getConnection() {
         return connection.get();
     }
 
+    /**
+     * Create a new SQLQuery instance with the given projection
+     *
+     * @param expr projection
+     * @param <T>
+     * @return select(expr)
+     */
+    public abstract <T> AbstractSQLQuery<T, ?> select(Expression<T> expr);
+
+    /**
+     * Create a new SQLQuery instance with the given projection
+     *
+     * @param exprs projection
+     * @return select(exprs)
+     */
+    public abstract AbstractSQLQuery<Tuple, ?> select(Expression<?>... exprs);
+
+    /**
+     * Create a new SQLQuery instance with the given projection
+     *
+     * @param expr distinct projection
+     * @param <T>
+     * @return select(distinct expr)
+     */
+    public abstract <T> AbstractSQLQuery<T, ?> selectDistinct(Expression<T> expr);
+
+    /**
+     * Create a new SQLQuery instance with the given projection
+     *
+     * @param exprs distinct projection
+     * @return select(distinct exprs)
+     */
+    public abstract AbstractSQLQuery<Tuple, ?> selectDistinct(Expression<?>... exprs);
+
+    /**
+     * Create a new SQLQuery instance with zero as the projection
+     *
+     * @return select(0)
+     */
+    public abstract AbstractSQLQuery<Integer, ?> selectZero();
+
+    /**
+     * Create a new SQLQuery instance with one as the projection
+     *
+     * @return select(1)
+     */
+    public abstract AbstractSQLQuery<Integer, ?> selectOne();
+
+    /**
+     * Create a new SQLQuery instance with the given projection and source
+     *
+     * @param expr query source and projection
+     * @param <T>
```

Comment:
AbstractSQLQuery has T for the projection type and Q for the concrete subclass type. Q is necessary for fluent method call invocations.